kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt

```diff
@@ -0,0 +1,373 @@
+                val isDateFilterSelected = dateSwitch.isChecked
+
+                // Retrieves from & to dates from the DatePicker
+                val fromDatePicker = fromDatePicker
+                val toDatePicker = toDatePicker
+                fromDate.set(fromDatePicker.year, fromDatePicker.month, fromDatePicker.dayOfMonth)
+                toDate.set(toDatePicker.year, toDatePicker.month, toDatePicker.dayOfMonth)
+
+                // Dismiss bottom sheet view
+                dialog.dismiss()
+
+                // Runs the query using the selected filters
+                runQuery(isDateFilterSelected)
+            }
+
+            // Dismiss bottom sheet view when cancel is clicked
+            cancelTv.setOnClickListener { dialog.dismiss() }
+        }
+
+        dialog.setCancelable(false)
+
+        // Sets bottom sheet content view to layout
+        dialog.setContentView(cqlFiltersLayoutBinding.root)
+
+        // Displays bottom sheet view
+        dialog.show()
+    }
+
+    /**
+     * Resets views in bottomSheet to default values.
+     */
+    private fun resetBottomSheetValues() {
+        cqlQueryListPosition = 4
+        maxFeatures = 3000
+        fromDate.set(2011, 5, 13)
+        toDate.set(2012, 0, 7)
+    }
+
+    /**
+     * Populates features from provided query parameters, and displays the result on the map.
+     * If [isDateFilterSelected] is true, query searches between fromDate-toDate
+     */
+    private fun runQuery(isDateFilterSelected: Boolean) {
+        val queryParameters = QueryParameters()
+
+        queryParameters.apply {
+            // Set the query parameter's where clause with the the selected query
+            whereClause = cqlQueryList[cqlQueryListPosition]
+
+            // Sets the max features to the number entered in the text field
+            maxFeatures = this@MainActivity.maxFeatures
+
+            // If date filter is selected, retrieve the date selected from the date picker
+            // and set it to the query parameters time extent
+            if (isDateFilterSelected) {
+                // set the query parameters time extent
+                timeExtent = TimeExtent(fromDate, toDate)
+            }
+        }
+
+        // Populate the table with the query, clear existing table entries
+        // and set the outfields parameter to null requests all fields
+        val result = ogcFeatureCollectionTable.populateFromServiceAsync(
+            queryParameters,
+            true,
+            null
+        )
+        result.addDoneListener {
+
+            // Create a new list to store returned geometries in
+            val featureGeometryList: MutableList<Geometry> = ArrayList()
+
+            // Iterate through each result to get its geometry and add it to the geometry list
+            result.get().iterator().forEach { feature ->
+                featureGeometryList.add(feature.geometry)
+                feature.geometry
```

Comment:
Do we need to capture feature.geometry here?

Reply:
My apologies, did not notice this until after it was merged to `v.next`. I have created #909 to address this.